### PR TITLE
Switch most calls to filepath.Walk to filepath.WalkDir

### PR DESCRIPTION
**Pull request by rhatdan**
It is faster then Walk, when you don't need to stat every
file and directory.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind @cevich This looks like the images does not have a newer golang?

**Comment by cevich** [comment]
> This looks like the images does not have a newer golang?

Newer golang is F36 only and in https://github.com/containers/podman/pull/13376

File: drivers/overlay/check_115.go [hunk 1]
```diff
@@ -1,8 +1,10 @@
+//go:build !go1.16
 // +build !go1.16
 
 package overlay
 
 import (
+	"io/fs"
```
Comment (nalind) [inline drivers/overlay/check_115.go:7]: This source file is a fallback for the sake of compilers earlier than 1.16, which don't have `io/fs` in their standard library.  You want to remove this file and merge `check_116.go` into `check.go` instead.

**Comment by rhatdan** [comment]
@cevich could you update the images for storage.  Podman's images do not want.  But I need newer versions of golang on f34.

**Comment by cevich** [comment]
> could you update the images for storage

Yep.

**Comment by cevich** [comment]
Let's see what https://github.com/containers/storage/pull/1182 do.

**Comment by TomSweeneyRedHat** [comment]
All kinds of test unhappiness @rhatdan

**Review by kolyshkin — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
Yup I am at a loss to why this works on Podman but not in storage.

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM
lint is complaining about not being able to find a package, plus you need to rebase, apparently.

**Comment by nalind** [comment]
The `lint` task in the Cirrus config specifies that it runs in `golang:1.15` on line 120 of `.cirrus.yml`, and `io/fs` was new in 1.16.

**Comment by cevich** [comment]
> The lint task in the Cirrus config specifies that it runs in golang:1.15 on line 120 of .cirrus.yml, and io/fs was new in 1.16.

IIUC, it's totally fine to just bump that up in the YAML, shouldn't affect anything other than that one lint task.

**Pull request merged**